ERCS/erc-7272.md

```diff
@@ -0,0 +1,221 @@
+        _;
+    }
+```
+
+The `verify` function calls the `AccessTokenVerifier` registered upon construction to verify the EAT's integrity.
+
+Optionally, implementers can decide to make EATs non-replayable. This what `_consumeAccessToken(...)` does. It marks an EAT as consumed to prevent its re-use:
+
+```
+function _consumeAccessToken(
+        uint8 v,
+        bytes32 r,
+        bytes32 s,
+        uint256 expiry
+    ) private {
+        bytes32 accessTokenHash = keccak256(abi.encodePacked(v, r, s, expiry));
+
+        _accessTokenUsed[accessTokenHash] = true;
+    }
+```
+
+#### AccessTokenVerifier
+The `verify()` function is part of the `AccessTokenVerifier` contract and is called by an `AccessTokenConsumer`.
+
+```
+    function verify(
+        AccessToken calldata token,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public view override returns (bool) {
+        address signer = _retrieveSignerFromToken(token, v, r, s);
+
+        // Verifies that the signer recovered from the token is a registered, active, expected
+        // issuer. How to register and manage signers onchain is out of scope in the context
+        // of ERC-7272.
+        return _isActiveIssuer[signer];
+    }
+```
+It takes a signature and an `AccessToken` as input, verifies that the token is not expired, attempts to recover the signer from the signature and the reconstructed EIP-712 digest, and verifies that the signer is a valid, expected signer. Please see the [reference implementation](../assets/erc-7272/AccessTokenVerifier.sol) for how this can be performed.
+
+## Rationale
+
+- Single-use. The reference implementation guarantees non-replayability of EATs. But other implementations might favor a different approach.
+
+- Use of EIP-712. By conforming to EIP-712, EATs are interoperable with existing Ethereum infrastructure, and developers can use them to create access controls with minimal modifications to their existing code. It also ensures that EATs issued are bound to a specific chain.
+
+- Zero-knowledge proofs. Using ZKPs comes at a cost, including added complexity. EATs are not much more than signed messages which are simpler to reason around. While `ecrecover` is available in any Ethereum smart contract out of the box, ZKPs come in different flavors which hinders interoperability.
+
+## Backwards Compatibility
+
+Any function can be gated with an EAT, apart from the special `receive` and `fallback` functions.
+
+
+## Reference Implementation
+
+Here's a reference implementation of the different smart contracts making up the EAT system onchain:
+
+- [IAccessTokenVerifier.sol](../assets//erc-7272/IAccessTokenVerifier.sol)
+- [AccessTokenVerifier.sol](../assets/erc-7272/AccessTokenVerifier.sol)
+- [AccessTokenConsumer.sol](../assets/erc-7272/AccessTokenConsumer.sol)
+
+## Security Considerations
+
+The security of the Ethereum Access Token (EAT) proposal depends on several factors:
+
+### Replay Attacks:
```

Comment:
```suggestion
### Replay Attacks
```

Reply:
Fixed in https://github.com/ethereum/ERCs/pull/92/commits/eba0d3558fdfeff91c1836d362f768d3a6785532